lib/file_diff.rb

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+
+class FileDiff
+  def self.diff(description1, description2, scope, path)
```

Comment:
The method is pretty short and straightforward, I don't really see how a refactoring could make it more readable or understandable. This metric probably is not that meaningful here because of the different prerequisite checks we need to do (scope extracted, file exists, file binary).